arviz/plots/backends/matplotlib/traceplot.py

```diff
@@ -124,6 +125,11 @@ def plot_trace(
 
     _, axes = plt.subplots(len(plotters), 2, squeeze=False, figsize=figsize, **backend_kwargs)
 
+    # Check the input for lines
+    if lines is not None:
+        if lines[0] not in [plotters[i][0] for i in range(len(plotters))]:
+            warnings.warn("A valid var_name should be provided")
```

Comment:
I don't think this will work at a quick glance (have not tried it yet though). Also now is probably a good time to fix docs, `lines` is a `list of tuple of (str, dict, float or array_like)`, so `lines[0]` will be a tuple. You should first get all var_names in a set (from plotters) and then check all var_names in lines tuples exist. It would be great to include _all_ incorrect variable names in warning message too.

Reply:
Yes, I wrote the code for keeping only a particular case in mind. This is failing for many other cases. I'll do the changes.